android/KMEA/app/src/main/java/com/tavultesoft/kmea/LanguageListActivity.java

```diff
@@ -324,340 +350,96 @@ protected static JSONObject getCachedJSONObject(Context context) {
     return jsonObj;
   }
 
-  private static void saveToCache(Context context, JSONObject jsonObj) {
-    ObjectOutput objOutput;
-    try {
-      // Save to cache file
-      objOutput = new ObjectOutputStream(new FileOutputStream(getCacheFile(context)));
-      objOutput.writeObject(jsonObj.toString());
-      objOutput.close();
-    } catch (Exception e) {
-      Log.e(TAG, "Failed to save to cache file. Error: " + e);
-    }
-  }
+  private static void showErrorDialog(Context context, String title, String message) {
+    AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(context);
 
-  @SuppressLint("SimpleDateFormat")
-  protected static Date getKeyboardModifiedDate(String keyboardID) {
-    Date date = null;
-    if (keyboardModifiedDates != null) {
-      String modDate = keyboardModifiedDates.get(keyboardID);
-      if (modDate != null && !modDate.isEmpty()) {
-        try {
-          SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZZZZ");
-          date = dateFormat.parse(modDate);
-        } catch (Exception e) {
-          date = null;
+    alertDialogBuilder.setTitle(title);
+    alertDialogBuilder
+      .setMessage(message)
+      .setCancelable(false)
+      .setPositiveButton(context.getString(R.string.label_close),new DialogInterface.OnClickListener() {
+        public void onClick(DialogInterface dialog,int id) {
+        if (dialog != null) {
+          dialog.dismiss();
         }
-      }
-    }
-    return date;
-  }
+        }
+      });
 
-  private class JSONParse extends AsyncTask<Void, Integer, JSONObject> {
```

Comment:
All `AsyncTask` code in this class was merged with that of `CloudRepository`'s `AsyncTask`, better centralizing our download code.

This was originally consulted during `CloudRepository`'s design to ensure a smooth process in that regard.